File: app/src/main/java/com/easy/lokalsolution/Fragment/SecondFragment.java
package com.easy.lokalsolution.Fragment;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.easy.lokalsolution.Class.QueryClass;
import com.easy.lokalsolution.Adapter.QueryAdapter;
import com.easy.lokalsolution.databinding.FragmentSecondBinding;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.Query;
import com.google.firebase.firestore.QuerySnapshot;

import java.util.ArrayList;


public class SecondFragment extends Fragment {

    public SecondFragment() {

    }

    FragmentSecondBinding binding;
    ArrayList<QueryClass> list = new ArrayList<>();
    QueryAdapter adapter;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        binding = FragmentSecondBinding.inflate(inflater, container, false);

        adapter = new QueryAdapter(getContext(), list);
        binding.queryrec.setAdapter(adapter);
        LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
        binding.queryrec.setLayoutManager(layoutManager);


        binding.shimmer.setVisibility(View.VISIBLE);
        binding.shimmer.startShimmer();
        FirebaseFirestore.getInstance().collection("Nanded")
                .document("NandedCity").collection("Query")
                .orderBy("time", Query.Direction.DESCENDING).get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
                    @Override
                    public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
                        list.clear();
                        for (DocumentSnapshot snapshot : queryDocumentSnapshots.getDocuments()) {
                            QueryClass data = snapshot.toObject(QueryClass.class);
                            list.add(data);
                        }
                        adapter.notifyDataSetChanged();
                        binding.shimmer.setVisibility(View.GONE);
                        binding.shimmer.stopShimmer();
                    }
                });

        binding.swiprefresh.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                binding.shimmer.setVisibility(View.VISIBLE);
                binding.shimmer.startShimmer();
                FirebaseFirestore.getInstance().collection("Nanded")
                        .document("NandedCity").collection("Query")
                        .orderBy("time", Query.Direction.DESCENDING).get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
                            @Override
                            public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
                                list.clear();
                                for (DocumentSnapshot snapshot : queryDocumentSnapshots.getDocuments()) {
                                    QueryClass data = snapshot.toObject(QueryClass.class);
                                    list.add(data);
                                }
                                adapter.notifyDataSetChanged();
                                binding.shimmer.setVisibility(View.GONE);
                                binding.shimmer.stopShimmer();
                            }
                        });

                binding.swiprefresh.setRefreshing(false);
                Toast.makeText(getContext(), "Data refresh", Toast.LENGTH_SHORT).show();

            }
        });

        binding.sort.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                binding.sortboxview.setVisibility(View.VISIBLE);
                binding.justview.setVisibility(View.VISIBLE);
            }
        });
        binding.justview.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                binding.sortboxview.setVisibility(View.GONE);
                binding.justview.setVisibility(View.GONE);
            }
        });
        binding.allsort.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                binding.shimmer.setVisibility(View.VISIBLE);
                binding.shimmer.startShimmer();
                FirebaseFirestore.getInstance().collection("Nanded")
                        .document("NandedCity").collection("Query")
                        .orderBy("time", Query.Direction.DESCENDING).get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
                            @Override
                            public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
                                list.clear();
                                for (DocumentSnapshot snapshot : queryDocumentSnapshots.getDocuments()) {
                                    QueryClass data = snapshot.toObject(QueryClass.class);
                                    list.add(data);
                                }
                                adapter.notifyDataSetChanged();
                                binding.shimmer.setVisibility(View.GONE);
                                binding.shimmer.stopShimmer();
                            }
                        });


                binding.sortboxview.setVisibility(View.GONE);
                binding.justview.setVisibility(View.GONE);
                Toast.makeText(getContext(), "All", Toast.LENGTH_SHORT).show();

            }
        });
        binding.rentout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                binding.shimmer.setVisibility(View.VISIBLE);
                binding.shimmer.startShimmer();
                FirebaseFirestore.getInstance().collection("Nanded")
                        .document("NandedCity").collection("Query")
                        .whereEqualTo("type","किरायाने देणे (Rent)")
                        .orderBy("time", Query.Direction.DESCENDING).get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
                            @Override
                            public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
                                list.clear();
                                for (DocumentSnapshot snapshot : queryDocumentSnapshots.getDocuments()) {
                                    QueryClass data = snapshot.toObject(QueryClass.class);
                                    list.add(data);
                                }
                                adapter.notifyDataSetChanged();
                                binding.shimmer.setVisibility(View.GONE);
                                binding.shimmer.stopShimmer();
                            }
                        });

                binding.sortboxview.setVisibility(View.GONE);
                binding.justview.setVisibility(View.GONE);
                Toast.makeText(getContext(), "किरायाने देणे", Toast.LENGTH_SHORT).show();

            }
        });  binding.rentin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                binding.shimmer.setVisibility(View.VISIBLE);
                binding.shimmer.startShimmer();
                FirebaseFirestore.getInstance().collection("Nanded")
                        .document("NandedCity").collection("Query")
                        .whereEqualTo("type","किरायाने पाहिजे (Rent)")
                        .orderBy("time", Query.Direction.DESCENDING).get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
                            @Override
                            public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
                                list.clear();
                                for (DocumentSnapshot snapshot : queryDocumentSnapshots.getDocuments()) {
                                    QueryClass data = snapshot.toObject(QueryClass.class);
                                    list.add(data);
                                }
                                adapter.notifyDataSetChanged();
                                binding.shimmer.setVisibility(View.GONE);
                                binding.shimmer.stopShimmer();
                            }
                        });

                binding.sortboxview.setVisibility(View.GONE);
                binding.justview.setVisibility(View.GONE);
                Toast.makeText(getContext(), "किरायाने पाहिजे", Toast.LENGTH_SHORT).show();

            }
        });
        binding.needbuy.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                binding.shimmer.setVisibility(View.VISIBLE);
                binding.shimmer.startShimmer();
                FirebaseFirestore.getInstance().collection("Nanded")
                        .document("NandedCity").collection("Query")
                        .whereEqualTo("type","विकत पाहिजे (Buy)")
                        .orderBy("time", Query.Direction.DESCENDING).get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
                            @Override
                            public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
                                list.clear();
                                for (DocumentSnapshot snapshot : queryDocumentSnapshots.getDocuments()) {
                                    QueryClass data = snapshot.toObject(QueryClass.class);
                                    list.add(data);
                                }
                                adapter.notifyDataSetChanged();
                                binding.shimmer.setVisibility(View.GONE);
                                binding.shimmer.stopShimmer();
                            }
                        });

                binding.sortboxview.setVisibility(View.GONE);
                binding.justview.setVisibility(View.GONE);
                Toast.makeText(getContext(), "विकत पाहिजे", Toast.LENGTH_SHORT).show();
            }
        });
        binding.sellout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                binding.shimmer.setVisibility(View.VISIBLE);
                binding.shimmer.startShimmer();
                FirebaseFirestore.getInstance().collection("Nanded")
                        .document("NandedCity").collection("Query")
                        .whereEqualTo("type","विक्री आहे (Sell)")
                        .orderBy("time", Query.Direction.DESCENDING).get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
                            @Override
                            public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
                                list.clear();
                                for (DocumentSnapshot snapshot : queryDocumentSnapshots.getDocuments()) {
                                    QueryClass data = snapshot.toObject(QueryClass.class);
                                    list.add(data);
                                }
                                adapter.notifyDataSetChanged();
                                binding.shimmer.setVisibility(View.GONE);
                                binding.shimmer.stopShimmer();
                            }
                        });

                binding.sortboxview.setVisibility(View.GONE);
                binding.justview.setVisibility(View.GONE);
                Toast.makeText(getContext(), "विक्री आहे", Toast.LENGTH_SHORT).show();
            }
        });
        binding.jobsort.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                binding.shimmer.setVisibility(View.VISIBLE);
                binding.shimmer.startShimmer();
                FirebaseFirestore.getInstance().collection("Nanded")
                        .document("NandedCity").collection("Query")
                        .whereEqualTo("type","नोकरी आहे (Job)")
                        .orderBy("time", Query.Direction.DESCENDING).get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
                            @Override
                            public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
                                list.clear();
                                for (DocumentSnapshot snapshot : queryDocumentSnapshots.getDocuments()) {
                                    QueryClass data = snapshot.toObject(QueryClass.class);
                                    list.add(data);
                                }
                                adapter.notifyDataSetChanged();
                                binding.shimmer.setVisibility(View.GONE);
                                binding.shimmer.stopShimmer();
                            }
                        });

                binding.sortboxview.setVisibility(View.GONE);
                binding.justview.setVisibility(View.GONE);
                Toast.makeText(getContext(), "नोकरी आहे", Toast.LENGTH_SHORT).show();
            }
        });
        binding.needjob.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                binding.shimmer.setVisibility(View.VISIBLE);
                binding.shimmer.startShimmer();
                FirebaseFirestore.getInstance().collection("Nanded")
                        .document("NandedCity").collection("Query")
                        .whereEqualTo("type","नोकरी पाहिजे (Job)")
                        .orderBy("time", Query.Direction.DESCENDING).get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
                            @Override
                            public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
                                list.clear();
                                for (DocumentSnapshot snapshot : queryDocumentSnapshots.getDocuments()) {
                                    QueryClass data = snapshot.toObject(QueryClass.class);
                                    list.add(data);
                                }
                                adapter.notifyDataSetChanged();
                                binding.shimmer.setVisibility(View.GONE);
                                binding.shimmer.stopShimmer();
                            }
                        });
                binding.sortboxview.setVisibility(View.GONE);
                binding.justview.setVisibility(View.GONE);
                Toast.makeText(getContext(), "नोकरी पाहिजे", Toast.LENGTH_SHORT).show();
            }
        });  binding.needperson.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                binding.shimmer.setVisibility(View.VISIBLE);
                binding.shimmer.startShimmer();
                FirebaseFirestore.getInstance().collection("Nanded")
                        .document("NandedCity").collection("Query")
                        .whereEqualTo("type","कामासाठी व्यक्ती पाहिजे (Need)")
                        .orderBy("time", Query.Direction.DESCENDING).get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
                            @Override
                            public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
                                list.clear();
                                for (DocumentSnapshot snapshot : queryDocumentSnapshots.getDocuments()) {
                                    QueryClass data = snapshot.toObject(QueryClass.class);
                                    list.add(data);
                                }
                                adapter.notifyDataSetChanged();
                                binding.shimmer.setVisibility(View.GONE);
                                binding.shimmer.stopShimmer();
                            }
                        });
                binding.sortboxview.setVisibility(View.GONE);
                binding.justview.setVisibility(View.GONE);
                Toast.makeText(getContext(), "कामासाठी व्यक्ती पाहिजे", Toast.LENGTH_SHORT).show();
            }
        });
        binding.othersort.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                binding.shimmer.setVisibility(View.VISIBLE);
                binding.shimmer.startShimmer();
                FirebaseFirestore.getInstance().collection("Nanded")
                        .document("NandedCity").collection("Query")
                        .whereEqualTo("type","वरील पैकी वेगळे (Other)")
                        .orderBy("time", Query.Direction.DESCENDING).get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
                            @Override
                            public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
                                list.clear();
                                for (DocumentSnapshot snapshot : queryDocumentSnapshots.getDocuments()) {
                                    QueryClass data = snapshot.toObject(QueryClass.class);
                                    list.add(data);
                                }
                                adapter.notifyDataSetChanged();
                                binding.shimmer.setVisibility(View.GONE);
                                binding.shimmer.stopShimmer();
                            }
                        });

                binding.sortboxview.setVisibility(View.GONE);
                binding.justview.setVisibility(View.GONE);
                Toast.makeText(getContext(), "other", Toast.LENGTH_SHORT).show();
            }
        });

        return binding.getRoot();
    }
}

File: app/src/main/java/com/easy/lokalsolution/Fragment/SecondFragment.kt
package com.easy.lokalsolution.Fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.easy.lokalsolution.Adapter.QueryAdapter
import com.easy.lokalsolution.Factory.QueryViewModelFactory
import com.easy.lokalsolution.Repository.QueryRepository
import com.easy.lokalsolution.ViewModel.QueryViewModel
import com.easy.lokalsolution.databinding.FragmentSecondBinding
import com.google.firebase.firestore.FirebaseFirestore

class SecondFragment() : Fragment() {
    var binding: FragmentSecondBinding? = null
     lateinit var viewModel: QueryViewModel
    lateinit var adapter: QueryAdapter
    public override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentSecondBinding.inflate(inflater, container, false)
        adapter = QueryAdapter(context, emptyList())
        binding!!.queryrec.adapter = adapter
         binding!!.queryrec.layoutManager = LinearLayoutManager(context)
        binding!!.shimmer.visibility = View.VISIBLE
        binding!!.shimmer.startShimmer()


        val firebase=FirebaseFirestore.getInstance()
        val repository=QueryRepository(firebase)
         viewModel= ViewModelProvider(this, QueryViewModelFactory(repository))[QueryViewModel::class.java]

        viewModel.allquery.observe(viewLifecycleOwner){
            adapter.updateData(it)

            binding!!.shimmer.visibility = View.GONE
            binding!!.shimmer.stopShimmer()
        }
        viewModel.loadAllQuery()









        binding!!.swiprefresh.setOnRefreshListener {
            binding!!.shimmer.visibility = View.VISIBLE
            binding!!.shimmer.startShimmer()
            viewModel.allquery.observe(viewLifecycleOwner){
                adapter.updateData(it)

                binding!!.shimmer.visibility = View.GONE
                binding!!.shimmer.stopShimmer()
            }
            viewModel.loadAllQuery()
            binding!!.swiprefresh.isRefreshing = false
            Toast.makeText(context, "Data refresh", Toast.LENGTH_SHORT).show()
        }


        binding!!.sort.setOnClickListener {
            binding!!.sortboxview.visibility = View.VISIBLE
            binding!!.justview.visibility = View.VISIBLE
        }
        binding!!.justview.setOnClickListener {
            binding!!.sortboxview.visibility = View.GONE
            binding!!.justview.visibility = View.GONE
        }



        binding!!.allsort.setOnClickListener {
            binding!!.shimmer.visibility = View.VISIBLE
            binding!!.shimmer.startShimmer()
            viewModel.allquery.observe(viewLifecycleOwner){
                adapter.updateData(it)

                binding!!.shimmer.visibility = View.GONE
                binding!!.shimmer.stopShimmer()
            }
            viewModel.loadAllQuery()
            binding!!.sortboxview.visibility = View.GONE
            binding!!.justview.visibility = View.GONE
            Toast.makeText(context, "All", Toast.LENGTH_SHORT).show()
        }




        binding!!.rentout.setOnClickListener {
            binding!!.shimmer.visibility = View.VISIBLE
            binding!!.shimmer.startShimmer()
            viewModel.allRentOutQuery.observe(viewLifecycleOwner){
                adapter.updateData(it)

                binding!!.shimmer.visibility = View.GONE
                binding!!.shimmer.stopShimmer()
            }
            viewModel.loadAllRentOutQuery()
            binding!!.sortboxview.visibility = View.GONE
            binding!!.justview.visibility = View.GONE
            Toast.makeText(context, "किरायाने देणे", Toast.LENGTH_SHORT).show()
        }




        binding!!.rentin.setOnClickListener {
            binding!!.shimmer.visibility = View.VISIBLE
            binding!!.shimmer.startShimmer()
            viewModel.allNeedByRentQuery.observe(viewLifecycleOwner){
                adapter.updateData(it)

                binding!!.shimmer.visibility = View.GONE
                binding!!.shimmer.stopShimmer()
            }
            viewModel.loadAllNeedByRentQuery()
            binding!!.sortboxview.visibility = View.GONE
            binding!!.justview.visibility = View.GONE
            Toast.makeText(context, "किरायाने पाहिजे", Toast.LENGTH_SHORT).show()
        }



        binding!!.needbuy.setOnClickListener {
            binding!!.shimmer.visibility = View.VISIBLE
            binding!!.shimmer.startShimmer()
            viewModel.allBuyQuery.observe(viewLifecycleOwner){
                adapter.updateData(it)

                binding!!.shimmer.visibility = View.GONE
                binding!!.shimmer.stopShimmer()
            }
            viewModel.loadAllBuyQuery()
            binding!!.sortboxview.visibility = View.GONE
            binding!!.justview.visibility = View.GONE
            Toast.makeText(context, "विकत पाहिजे", Toast.LENGTH_SHORT).show()
        }




        binding!!.sellout.setOnClickListener {
            binding!!.shimmer.visibility = View.VISIBLE
            binding!!.shimmer.startShimmer()
            viewModel.allSellQuery.observe(viewLifecycleOwner){
                adapter.updateData(it)

                binding!!.shimmer.visibility = View.GONE
                binding!!.shimmer.stopShimmer()
            }
            viewModel.loadAllSellQuery()
            binding!!.sortboxview.visibility = View.GONE
            binding!!.justview.visibility = View.GONE
            Toast.makeText(context, "विक्री आहे", Toast.LENGTH_SHORT).show()
        }




        binding!!.jobsort.setOnClickListener {
            binding!!.shimmer.visibility = View.VISIBLE
            binding!!.shimmer.startShimmer()
            viewModel.allJobQuery.observe(viewLifecycleOwner){
                adapter.updateData(it)

                binding!!.shimmer.visibility = View.GONE
                binding!!.shimmer.stopShimmer()
            }
            viewModel.loadAllJobQuery()
            binding!!.sortboxview.visibility = View.GONE
            binding!!.justview.visibility = View.GONE
            Toast.makeText(context, "नोकरी आहे", Toast.LENGTH_SHORT).show()
        }



        binding!!.needjob.setOnClickListener {
            binding!!.shimmer.visibility = View.VISIBLE
            binding!!.shimmer.startShimmer()
            viewModel.allNeedJobQuery.observe(viewLifecycleOwner){
                adapter.updateData(it)

                binding!!.shimmer.visibility = View.GONE
                binding!!.shimmer.stopShimmer()
            }
            viewModel.loadAllNeedJobQuery()
            binding!!.sortboxview.visibility = View.GONE
            binding!!.justview.visibility = View.GONE
            Toast.makeText(context, "नोकरी पाहिजे", Toast.LENGTH_SHORT).show()
        }



        binding!!.needperson.setOnClickListener {
            binding!!.shimmer.visibility = View.VISIBLE
            binding!!.shimmer.startShimmer()
            viewModel.allNeedWorkerQuery.observe(viewLifecycleOwner){
                adapter.updateData(it)

                binding!!.shimmer.visibility = View.GONE
                binding!!.shimmer.stopShimmer()
            }
            viewModel.loadAllNeedWorkerQuery()
            binding!!.sortboxview.visibility = View.GONE
            binding!!.justview.visibility = View.GONE
            Toast.makeText(context, "कामासाठी व्यक्ती पाहिजे", Toast.LENGTH_SHORT).show()
        }



        binding!!.othersort.setOnClickListener {
            binding!!.shimmer.visibility = View.VISIBLE
            binding!!.shimmer.startShimmer()
            viewModel.allOtherQuery.observe(viewLifecycleOwner){
                adapter.updateData(it)

                binding!!.shimmer.visibility = View.GONE
                binding!!.shimmer.stopShimmer()
            }
            viewModel.loadAllOtherQuery()
            binding!!.sortboxview.visibility = View.GONE
            binding!!.justview.visibility = View.GONE
            Toast.makeText(context, "other", Toast.LENGTH_SHORT).show()
        }
        return binding!!.root
    }
}